src/dawn/IIR/StencilMetaInformation.cpp

```diff
@@ -254,6 +456,21 @@ void StencilMetaInformation::removeAccessID(int AccessID) {
   }
 }
 
+StencilMetaInformation::StencilMetaInformation(const sir::GlobalVariableMap& globalVariables) {
```

Comment:
I would say yes, that one should be always required, it can be empty though. Having a default ctr can lead to the global variable related maps if you forget to construct them later on. Building always at ctr time follows the RAII

Reply:
makes sense.